velox/exec/tests/PlanNodeSerdeTest.cpp

```diff
@@ -100,6 +100,35 @@ TEST_F(PlanNodeSerdeTest, crossJoin) {
   testSerde(plan);
 }
 
+TEST_F(PlanNodeSerdeTest, nestedLoopJoin) {
+  auto left = makeRowVector(
+      {"t0", "t1", "t2"},
+      {
+          makeFlatVector<int32_t>({1, 2, 3}),
+          makeFlatVector<int64_t>({10, 20, 30}),
+          makeFlatVector<bool>({true, true, false}),
+      });
+
+  auto right = makeRowVector(
+      {"u0", "u1", "u2"},
+      {
+          makeFlatVector<int32_t>({1, 2, 3}),
+          makeFlatVector<int64_t>({10, 20, 30}),
+          makeFlatVector<bool>({true, true, false}),
+      });
+
+  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+  auto plan =
+      PlanBuilder(planNodeIdGenerator)
+          .values({left})
+          .nestedLoopJoin(
```

Comment:
Would you add a test with non-default join type to make sure it is preserved during serde?